documentation/configuration_files/service.md

```diff
@@ -41,3 +41,36 @@ dependency-data:
 
 * `remote.cpu`: Number of CPU units to reserve for service container (see "cpu" under [Container Definitions/Environment](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_environment))
 * `remote.memeory`: The hard limit (in MiB) of memory to allocate to the service container (see "memory" under [Container Definitions](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definitions))
+
+# Service types
+
+## Public services
+Public services expose an external and internal http endpoint. Container ports must be listed in `development.port` and `production.port`.
+
+Endpoints available in local development:
+  - `#{SERVICE_ROLE}_EXTERNAL_ORIGIN: http://localhost:#{host-port}`: The external endpoint at which a public service can be reached. Exosphere automatically picks an available host port and binds it to the specified service container port.
+  - `#{SERVICE_ROLE}_INTERNAL_ORIGIN: http://#{service-role}:#{container-port}`: The internal endpoint at which a service can be reached. Used for internal communication with other services.
+
+Endpoints availabe in deployment:
+  - `#{SERVICE_ROLE}_EXTERNAL_ORIGIN: http://#{service-url}`: The load-balanced external endpoint, as defined by the URL listed in the `remote.envorinments.#{remote-environment-id}` block of `service.yml`. Terraform manages creation of these records in route53.
+  - `#{SERVICE_ROLE}_INTERNAL_ORIGIN: http://#{service-role}.#{remote-environment-id}-#{app-name}.local`: The internal endpoint at which a service can be reached. Terraform manages creation of these records in route53. Used for internal communication with other services.
+
+
+## Worker services
+Worker services have the option of exposing an internal non-http, non-load-balanced endpoint. If a worker port is exposed, they must be listed in `development.port` and `production.port`.
+Additionally, the proper `update-route53` binary must be downloaded from Exosphere's [release](https://github.com/Originate/exosphere/releases), copied into the worker service's Docker container and ran as part of that service's start-up script.
```

Comment:
I think we should build something into the binary so `update-route53` does nothing when the application is being running locally otherwise I expect this to break things if running the production image locally

Reply:
What I can think of is passing an environment variable `$ENV` into the binary that determines whether it actually executes. Do you have any ideas?